lib/structuraid_core/designers/footings/steps/centric_isolated/set_initial_longitudinal_reinforcement.rb

```diff
@@ -0,0 +1,146 @@
+require 'interactor'
+require 'matrix'
+
+module StructuraidCore
+  module Designers
+    module Footings
+      module Steps
+        module CentricIsolated
+          # Runs the structural bending analysis and design for a centric isolated footing, generating the required reinforcement ratio and checking that the provided one is enough
+          class SetInitialLongitudinalReinforcement
+            include Interactor
+
+            POSSIBLE_SECCTIONS = %i[length_1 length_2].freeze
+
+            # @param footing [StructuraidCore::Elements::Footing] The footing to be designed
+            # @param load_scenario [StructuraidCore::Loads::Scenarios::Footings::CentricIsolated] The load scenario to be considered
+            # @param analysis_direction [Symbol] The direction for which the analysis has to be run. Should be either :length_1 or :length2
+            # @param design_code [StructuraidCore::DesignCodes] The design code to be used
+            # @param steel [StructuraidCore::Materials::Steel] The rebar's material
+            # @param support_type [Symbol or String] The support type: :over_soil or :over_piles
+            def call
+              return unless footing.reinforcement_ratio(direction: analysis_direction, above_middle: true).zero?
+
+              add_reinforcement
+            rescue Errors::Designers::DesignStepError => e
+              context.fail!(message: e.message)
+            end
+
+            private
+
+            def add_reinforcement
+              return add_reinforcement_to_footing unless optimized_rebar.result_code == critical_result_code
+
+              raise Errors::Designers::DesignStepError.new(optimized_rebar.result_code, "rebar couldm't be found")
```

Comment:
I don't think we need this error. Just call  `context.fail!` whenever you have an error to throw that breaks the interactor

Reply:
Done!